tests/test_selector_csstranslator.py

```diff
@@ -196,3 +201,10 @@ def test_nested_selector(self):
             self.sel.css("div").css("area:last-child").extract(),
             ['<area shape="default" id="area-nohref">'],
         )
+
+    @pytest.mark.xfail(
+        parse_version(cssselect.__version__) < parse_version("1.2.0"),
+        reason="Support added in cssselect 1.2.0",
+    )
+    def test_pseudoclass_has(self):
+        self.assertEqual(self.x("p:has(b)::text"), ["lorem ipsum text"])
```

Comment:
Would this test fail if not for the changes to `join`?

Reply:
Yes, with an aforementioned TypeError.